app/upgrade.go

```diff
@@ -105,7 +109,7 @@ func (app *App) registerPampasUpgradeHandler() {
 	app.UpgradeKeeper.SetUpgradeInitializer(upgradetypes.Pampas,
 		func() error {
 			app.Logger().Info("Init Pampas upgrade")
-
+			app.setPaymentConsensusV2()
```

Comment:
because `registerNagquUpgradeHandler` already do `app.setPaymentConsensusV2()`, wont them duplicate?

Reply:
The issue is that: for example,  we state sync from a height `H` ( Nagqu  < `H` < Pampas),  when running to Pampas upgrade, if we do not set consensus version to `2` , when running migrations it will read `1` from app and save `1` to db, then it will lead to app hash mismatch issue.